### DISCUSSION TITLE
Support field init shorthands?

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#8379] [Ideas]
Rust allows:

```rust
struct Dog {
    name: String,
    breed: String,
    age: u32,
}

fn instead_of_this(name: String, breed: String) -> Dog {
    Dog {
        name: name,
        breed: breed,
        age: 0,
    }
}

fn do_this(name: String, breed: String) -> Dog {
    Dog {
        name,
        breed,
        age: 0,
    }
}
```

The key is that `name` is used both as the field name and the local variable that is referenced.

(Rust even allows the fields to occur out-of-order, but we do not have to support that).

**Comment by magnus-madsen** [comment]
@dghosef @mlutze @JonathanStarup CC

**Reply by JonathanStarup** [reply]
If we have it for structs it should also be usable in records, e.g. `{x, y}` means `{x = x, y = y}`

**Comment by dghosef** [comment]
I feel that since we are requiring the ordering, we can make the name optional.

**Reply by JonathanStarup** [reply]
I don't know if we want to commit to that
